nav2_behavior_tree/plugins/condition/is_path_valid_condition.cpp

```diff
@@ -0,0 +1,52 @@
+
+#include <string>
+#include "nav2_behavior_tree/plugins/condition/is_path_valid_condition.hpp"
+
+
+namespace nav2_behavior_tree
+{
+ValidPathCondition::ValidPathCondition(
+  const std::string & condition_name,
+  const BT::NodeConfiguration & conf)
+: BT::ConditionNode(condition_name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+}
+
+BT::NodeStatus ValidPathCondition::tick()
+{
+
+  // Grab the path
+  nav_msgs::msg::Path path;
+  getInput("path", path);
+
+  auto request = std::make_shared<nav2_msgs::srv::IsPathValid::Request>();
+
+  request->path = path; 
+
+  //To be implemented
+  auto result = client->async_send_request(request);
+  // Wait for the result.
+  if (rclcpp::spin_until_future_complete(node_, result) ==
```

Comment:
See the timeout logic for action nodes -- if we spin, we should have a timeout in case of networking failures (don't want to block indefinitely)

Reply:
Is "RUNNING" a valid state for condition nodes? If so, might want to consider also the timeout based on the behavior tree tick rate so that we can tick the tree at a constant rate and spin again until the final timeout is reached in behavior tree spin rate increments.

Reply:
Conditional BT nodes can only return SUCCESS or FAILURE. Would this better suited as action node to allow for RUNNING to be returned when waiting for the service(or fail)?

Reply:
This should be fast, but once you have it implemented, it would be good to benchmark how long it takes. If its in line with the tick rate, lets just keep it a condition. I think semantically that makes more sense. If it takes too long though, we might want to weight the benefits / costs.